runtime_lib/controller/main.cpp

```diff
@@ -820,6 +820,48 @@ void xaie_l2_dma_init(int col) {
   xaie::out32(xaie::getTileAddr(col, 0) + 0x00033008, 0xFF);
 }
 
+#ifdef ARM_CONTROLLER
+
+// Defining the NPI base and registers we use to reset the array
+uint64_t npi_base = 0xF70A0000UL;
+#define NPI_MASK_REG 0x0
+#define NPI_VAL_REG 0x4
+#define NPI_LOCK_REG 0xC
+void xaie_array_reset() {
+
+  // Getting a pointer to NPI
+  volatile uint32_t *npib = (volatile uint32_t *)(npi_base);
+
+  // Performing array reset sequence
+  air_printf("Starting array reset\r\n");
+  npib[NPI_LOCK_REG >> 2] = 0xF9E8D7C6;
```

Comment:
A lot of magical incantation for an open-source project. Perhaps adding some comments at least about what it means?

Reply:
Don't want to go into too much detail of the content of the NPI registers, but added some general comments on when the reset is happening in [173d9da](https://github.com/Xilinx/mlir-air/pull/253/commits/173d9da6995d41a594bd9dc3e6d8972005afa95f)